docs/source/_notebooks/xgboost-titanic.rst

```diff
@@ -0,0 +1,2468 @@
+ensemble classifier (XGBoost in this case, but it also works for tree
+ensembles from scikit-learn and for regression). We will use `Titanic
+dataset <https://www.kaggle.com/c/titanic/data>`__, which is small and
+has not too many features, but is still rich enough.
+
+1. Training data
+----------------
+
+Let's start by loading the data:
+
+.. code:: python
+
+    import csv
+    import numpy as np
+    
+    with open('titanic-train.csv', 'rt') as f:
+        data = list(csv.reader(f))
+
+Variable descriptions:
+
+-  **Survival:** Survival (0 = No; 1 = Yes)
+-  **Pclass:** Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd)
+-  **Name:** Name
+-  **Sex:** Sex
+-  **Age:** Age
+-  **Sibsp:** Number of Siblings/Spouses Aboard
+-  **Parch:** Number of Parents/Children Aboard
+-  **Ticket:** Ticket Number
+-  **Fare:** Passenger Fare
+-  **Cabin:** Cabin
+-  **Embarked:** Port of Embarkation (C = Cherbourg; Q = Queenstown; S =
+   Southampton)
+
+Next, shuffle data and separate features from what we are trying to
+predict: survival.
+
+.. code:: python
+
+    from pprint import pprint
+    from sklearn.utils import shuffle
+    from sklearn.model_selection import train_test_split
+    
+    feature_names = data[0][2:]
+    _all_xs = [dict(zip(feature_names, row[2:])) for row in data[1:]]
+    _all_ys = np.array([int(row[1]) for row in data[1:]])
+    
+    all_xs, all_ys = shuffle(_all_xs, _all_ys, random_state=0)
+    train_xs, valid_xs, train_ys, valid_ys = train_test_split(
+        all_xs, all_ys, test_size=0.25, random_state=0)
+    print('{} items total, {:.1%} true'.format(len(all_xs), np.mean(all_ys)))
+    pprint(valid_xs[:2])
+
+
+.. parsed-literal::
+
+    891 items total, 38.4% true
+    [{'Age': '19',
+      'Cabin': '',
+      'Embarked': 'S',
+      'Fare': '10.1708',
+      'Name': 'Dakic, Mr. Branko',
+      'Parch': '0',
+      'Pclass': '3',
+      'Sex': 'male',
+      'SibSp': '0',
+      'Ticket': '349228'},
+     {'Age': '19',
+      'Cabin': '',
+      'Embarked': 'Q',
+      'Fare': '7.8792',
+      'Name': 'Devaney, Miss. Margaret Delia',
+      'Parch': '0',
+      'Pclass': '3',
+      'Sex': 'female',
+      'SibSp': '0',
+      'Ticket': '330958'}]
+
+
+We do just minimal preprocessing: convert obviously contiuous ``Age``
+and ``Fare`` variables to floats, and ``SibSp``, ``Parch`` to integers.
+``Age`` can be missing, we default to 0 and will tell XGBoost to treat
+it as missing later.
+
+.. code:: python
+
+    for x in all_xs:
+        x['Age'] = float(x['Age'] or 0)
+        x['Fare'] = float(x['Fare'])
+        x['SibSp'] = int(x['SibSp'])
+        x['Parch'] = int(x['Parch'])
+
+2. Simple XGBoost classifier
+----------------------------
+
+Let's first build a very simple classifier with ``XGBClassifier`` and
+``sklearn.feature_extraction.DictVectorizer``, and check it's accuracy
+with cross-validation:
+
+.. code:: python
+
+    import warnings
+    warnings.filterwarnings('ignore', category=DeprecationWarning) 
```

Comment:
I think a comment about the purpose of this .filterwarnings could be nice here ("xgboost <= 0.6a2 shows a warning when used with scikit-learn 0.18+" or something like that)